Batteries/Data/Nat/Digits.lean

```diff
@@ -0,0 +1,36 @@
+/-
+Copyright (c) 2025 Marcus Rossel. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Marcus Rossel
+-/
+
+namespace Nat
+
+@[simp]
+theorem isDigit_digitChar : n.digitChar.isDigit = decide (n < 10) :=
+  match n with
+  | 0 | 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 => by simp [digitChar]
+  | _ + 10 => by
+    simp only [digitChar, ↓reduceIte, Nat.reduceEqDiff]
+    (repeat' split) <;> simp
+
+private theorem isDigit_of_mem_toDigitsCore_10
+    (hf : n < fuel) (hc : c ∈ cs → c.isDigit) (h : c ∈ toDigitsCore 10 fuel n cs) :
+    c.isDigit := by
+  induction fuel generalizing n cs <;> rw [toDigitsCore] at h
+  next => contradiction
+  next fuel ih =>
+    split at h
+    case' isFalse => apply ih (by omega) (fun h => ?_) h
+    all_goals
+      cases h
+      next => simp [mod_lt]
+      next hm => exact hc hm
+
+theorem isDigit_of_mem_toDigits_10 (h : c ∈ toDigits 10 n) : c.isDigit :=
```

Comment:
This holds more generally as
```suggestion
theorem isDigit_of_mem_toDigits (k) (hk : k ≤ 10) (h : c ∈ toDigits k n) : c.isDigit :=
```
right?

Reply:
I just noticed, the implementation for base `1` is in fact incorrect. E.g. `toDigits 1 3` returns `['0', '0', '0', '0']`. It only terminates in the first place, because the fuel counts down.

Reply:
This is being discussed at https://leanprover.zulipchat.com/#narrow/channel/270676-lean4/topic/.60Nat.2EtoDigits.60.20base.201/with/523028786 now.